dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -604,5 +622,128 @@ public String getUrl(Contentlet contentlet) {
+
+        final File file = new File(directory,String.format("%s.serialized",System.currentTimeMillis()));
+        file.createNewFile();
+
+        final Contentlet original = serializationTestCase.contentlet;
+        serializeContentlet(serializationTestCase.contentlet, file);
+        final Contentlet copy = readSerializedContentlet(file);
+        validateTransformation(original, copy);
+        final AssertionStrategy assertionStrategy = serializationTestCase.assertionStrategy;
+        if(null != assertionStrategy) {
+            assertionStrategy.apply(original, copy);
+        }
+    }
+
+    private static class SerializationTestCase {
+
+        final Contentlet contentlet;
+        final AssertionStrategy assertionStrategy;
+
+        SerializationTestCase(final Contentlet contentlet,
+                final AssertionStrategy assertionStrategy) {
+            this.contentlet = contentlet;
+            this.assertionStrategy = assertionStrategy;
+        }
+    }
+
+    @DataProvider
+    public static Object[] listSerializeTestCases() throws Exception {
+
+        final FileAssetAPI assetAPI = APILocator.getFileAssetAPI();
+        final HTMLPageAssetAPI pageAssetAPI = APILocator.getHTMLPageAssetAPI();
+
+        return new Object[]{
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getFileAssetSVGContent(true, langId);
+                    return assetAPI.fromContentlet(contentlet);
+                }).get(), fileAssetValidation),
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getFileAssetContent(true, langId);
+                    return assetAPI.fromContentlet(contentlet);
+                }).get(), fileAssetValidation),
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getPageContent(true, langId);
+                    return pageAssetAPI.fromContentlet(contentlet);
+                }).get(), pageValidation),
+                new SerializationTestCase(TestDataUtils.getDotAssetLikeContentlet(), null),
+                new SerializationTestCase(TestDataUtils.getBlogContent(true, langId), null)
+        };
+    }
+
+    private void validateTransformation(final Contentlet original, final Contentlet copy){
+        final DotContentletTransformer transformer1 = new DotTransformerBuilder().defaultOptions().content(original).build();
+        final Map<String, Object> preSerializationMap = transformer1.toMaps().get(0);
+
+        final DotContentletTransformer transformer2 = new DotTransformerBuilder().defaultOptions().content(copy).build();
+        final Map<String, Object> postSerializationMap = transformer2.toMaps().get(0);
+        Assert.assertEquals(preSerializationMap, postSerializationMap);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5883765)